### PR TITLE
Document how to set up a dev environment for recipe-client-addon.

### PR DESCRIPTION
**Pull request by Osmose**

### CODE REVIEW

File: docs/dev/recipe-client-addon/workflow.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+   ./bin/update-mozilla-central.sh /path/to/mozilla-central
+
+.. _recipe-client-tests:
+
+Writing and Running Tests
+-------------------------
+The recipe client tests are designed to be run using mozilla-central's test
+frameworks. There are multiple test suites:
+
+* xpcshell_ tests, located in ``/recipe-client-addon/test/unit``, are run in a
```
Comment (mythmon) [inline docs/dev/recipe-client-addon/workflow.rst]: Are we going to ditch xpcshell tests? Or are we going to keep having both? If we are going to ditch them, we should make that clear here.

Comment (Osmose) [inline docs/dev/recipe-client-addon/workflow.rst]: Ditch. I'll add a note.

**Comment by andymikulski** [comment]
Going to run through this locally to review. Thanks for putting this together!

File: docs/dev/recipe-client-addon/workflow.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  JavaScript shell. Some modules do not work in this environment because it
+  lacks things like access to a browser window, but the test output is more
+  readable, and executing tests does not require focus.
+
+* mochitest_ tests, located in ``/recipe-client-addon/test/browser``, are run in
+  a webpage that can import and run chrome code. Mochitest can test pretty much
+  anything, but running the test requires you to focus the browser windows that
+  the test framework opens, and the test output is sometimes difficult to read.
+
+To run the tests, you must sync your code do mozilla-central, and use the
```
Comment (andymikulski) [inline docs/dev/recipe-client-addon/workflow.rst]: typo: `code do mozilla` -> `code to mozilla`

File: docs/dev/recipe-client-addon/workflow.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+You can also perform limited linting using eslint_ directly. To do this, you
+must install the development dependencies from
+``/recipe-client-addon/package.json`` in the same environment that eslint is
+installed within to get the eslint plugins we rely on:
+
+.. code-block:: bash
+
+   cd recipe-client-addon
+   npm install
+   ./node_modules/.bin/eslint
```
Comment (andymikulski) [inline docs/dev/recipe-client-addon/workflow.rst]: I can't get this to work locally, have you tested this yourself? Few things:
- Running just `.../eslint` will bring up the help menu
- Running a command like `.../eslint ./lib/PreferenceExperiments.jsm -c .eslintrc.js` doesn't do anything (drops a line to indicate something is running, but there is nothing outputted to console)

Comment (Osmose) [inline docs/dev/recipe-client-addon/workflow.rst]: Ohhhhhh, there's a bug with .eslintignore, I can fix that.

**Comment by Osmose** [comment]
Updated with fixes. Running eslint manually reveals some lint errors that are not caught by `mach eslint`; @mythmon is about to put up a separate PR to fix this, so I don't think we should block on those errors existing.

File: docs/dev/recipe-client-addon/workflow.rst (hunk 3 above)
Comment (andymikulski) [inline docs/dev/recipe-client-addon/workflow.rst]: I still have the issue of eslint hanging after running the command. Maybe I'm missing a PR that already landed on master or something.

Comment (Osmose) [inline docs/dev/recipe-client-addon/workflow.rst]: It should be included in this PR. If you switch to this branch, do you have a `.eslintignore` file at the root of your repo? Also, what's the version number in your `recipe-client-addon/node_modules/eslint-plugin-mozilla/package.json` file?

**Pull request merged**